src/main/java/io/jenkins/plugins/sample/OnboardingConfiguration.java

```diff
@@ -0,0 +1,181 @@
+        } else {
+            if (!PluginHelper.isMatch(username, USERNAME_REGEX_PATTERN)) {
+                return FormValidation.error("Invalid Username format");
+            }
+            return FormValidation.ok();
+        }
+    }
+
+    /**
+     * A field validator for name field (plugin name) on the optional view on the Onboarding Plugin Section
+     * @param name
+     *        name field name (plugin name) to validate
+     * @return
+     *       FormValidation object
+     */
+    public FormValidation doCheckName(@QueryParameter String name) {
+        if (name.isEmpty()) {
+            return FormValidation.error("Please enter your name");
+        } else {
+            if (!PluginHelper.isMatch(name, PLUGIN_NAME_REGEX_PATTERN)) {
+                return FormValidation.error("Invalid name format");
+            }
+            return FormValidation.ok();
+        }
+    }
+
+    /**
+     * This method validates the name and username parameters based on validation regexes
+     * {@link #PLUGIN_NAME_REGEX_PATTERN} and {@link #USERNAME_REGEX_PATTERN}
+     * It just overrides the {@link jenkins.model.GlobalConfiguration#configure(StaplerRequest2, JSONObject)}} method
+     * to validate name (plugin name) and username fields when the Save or Apply Button is clicked on System Configuration page
+     * @param req
+     *      The stapler request object
+     * @param json
+     *      The JSON object that captures the configuration data for this {@link hudson.model.Descriptor}.
+     *      See <a href="https://www.jenkins.io/doc/developer/forms/structured-form-submission/">the developer documentation</a>.
+     * @return boolean If validation for the fields (name and username) is successfull, return true else return false
+     * @throws FormException
+     */
+    @Override
+    public boolean configure(StaplerRequest2 req, JSONObject json) throws FormException {
+        String pluginName = json.getString("name");
+        String userName = json.getString("username");
+
+        if (pluginName != null || userName != null) {
+            if (!PluginHelper.isMatch(pluginName, PLUGIN_NAME_REGEX_PATTERN)) {
+                return false;
+            }
+            if (!PluginHelper.isMatch(userName, USERNAME_REGEX_PATTERN)) {
+                return false;
+            }
+            req.bindJSON(this, json);
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    /**
+     * This method validates the url , username and password fields in the optional view on the Onboarding Plugin Section
+     * @param url
+     *      url to send username and password to
+     * @param username
+     *      username field
+     * @param password
+     *      password field
+     * @return
+     *      FormValidation object
+     * @throws IOException
+     * @throws ServletException
+     */
+    @POST
+    public FormValidation doTestConnection(
```

Comment:
This method is missing a permission check 🙀 

Time to add one as described on https://www.jenkins.io/doc/developer/security/form-validation/

Reply:
I see no permission checks in the current state of the PR…?

Note that setters cannot be directly invoked, but anyone with Overall/Read permission can access the URL for `doTestConnection` that powers the form validation. That's why it needs a permission check.